arviz/tests/conftest.py

```diff
@@ -0,0 +1,38 @@
+"""Configuration for test suite."""
+# pylint: disable=redefined-outer-name
+import os
+import pytest
+
+
+def pytest_addoption(parser):
+    """Definition for command line option to save figures from tests."""
+    parser.addoption("--save", nargs="?", const="test_images",
+                     help="Save images rendered by plot")
+
+
+@pytest.fixture(scope="session")
+def save_figs(request):
+    """Enable command line switch for saving generation figures upon testing."""
+    fig_dir = request.config.getoption("--save")
+
+    if fig_dir is not None:
+        # Try creating directory if it doesn't exist
+        print("Saving generated images in {}".format(fig_dir))
+        try:
+            os.mkdir(fig_dir)
+            print("Directory {} created".format(fig_dir))
+
+        except FileExistsError:
+            print("Directory {} exists".format(fig_dir))
+
+        # Clear all files from the directory
+        for file in os.listdir(fig_dir):
+            full_path = os.path.join(fig_dir, file)
+
+            try:
+                os.remove(full_path)
+
+            except OSError:
+                print("Failed to remove {}".format(full_path))
+
```

Comment:
Images are not locked, so overwriting is fine. Maybe catch exception when saving fails and then retry n-times (use maybe date-to-milliseconds as extra identifier)

Reply:
I think images lock on Windows if the user has it open in in Windows photo viewer but I could be wrong.

Reply:
Oh I also decided to remove the images up front rather than overwrite during testing.

The reason for this is that if I run a full test suite ~100 images will be generated, but I run one test afterward only one image will be generated and overwritten, and the other 99 would just remain in the directory.
This may lead to confusion because one directory could contain renderings from different version of ArviZ.

Figured it was cleaner to always clear the directory at the start of each test run.

Does this pattern make sense?

Reply:
Yes.

You can not remove file that is open, but you can overwrite it.